tikv/mvcc.go

```diff
@@ -682,17 +697,23 @@ func (store *MVCCStore) prewritePessimistic(reqCtx *requestCtx, mutations []*kvr
 
 func (store *MVCCStore) prewriteMutations(reqCtx *requestCtx, mutations []*kvrpcpb.Mutation,
 	req *kvrpcpb.PrewriteRequest, items []*badger.Item) error {
-	batch := store.dbWriter.NewWriteBatch(req.StartVersion, 0, reqCtx.rpcCtx)
 	var minCommitTS uint64
-	if req.UseAsyncCommit {
+	if req.UseAsyncCommit || req.TryOnePc {
 		// Get minCommitTS for async commit protocol. After all keys are locked in memory lock.
 		physical, logical, tsErr := store.pdClient.GetTS(context.Background())
 		if tsErr != nil {
 			return tsErr
 		}
 		minCommitTS = uint64(physical)<<18 + uint64(logical)
-		reqCtx.asyncMinCommitTS = minCommitTS
+		if req.UseAsyncCommit {
+			reqCtx.asyncMinCommitTS = minCommitTS
+		} else if !req.TryOnePc {
+			log.Fatal("unreachable")
+		}
 	}
+
+	keys := make([][]byte, 0, len(mutations))
+	locks := make([]*mvcc.MvccLock, 0, len(mutations))
 	for i, m := range mutations {
 		if m.Op == kvrpcpb.Op_CheckNotExists {
 			continue
```

Comment:
unfold the code below, we don't need to set the MinCOmmitTs in the loop.

Reply:
It looks needed for `buildPrewriteLock`.

Reply:
Yes, but we can put it out of the loop.